scripts/apps/authoring-react/authoring-integration-wrapper.tsx

```diff
@@ -359,6 +359,37 @@ export class AuthoringIntegrationWrapper extends React.PureComponent<IPropsWrapp
                             },
                             retrieveStoredValue: (item: IArticle, fieldId) => item.extra?.[fieldId] ?? null,
                         }}
+                        customHeader={((exposed) => {
+                            const getProfileAndReinitialize = (item: IArticle) =>
+                                this.props.authoringStorage.getContentProfile(
+                                    item,
+                                    exposed.fieldsAdapter,
+                                ).then((profile) => {
+                                    exposed.reinitialize(item, profile);
+                                });
+
+                            return [{
+                                component: ({entity}) => (
+                                    <div className="authoring-header__general-info">
+                                        <ContentProfileDropdown
+                                            item={entity}
+                                            reinitialize={(item) => {
+                                                if (exposed.hasUnsavedChanges()) {
```

Comment:
It'd be cleaner if `getProfileAndReinitialize` would only be used once. This way it is also clear that it will get called regardless of the condition.

```
const handledChanges = exposed.hasUnsavedChanges()
    ? exposed.handleUnsavedChanges()
    : Promise.resolve();

handledChanges.then(() => {
    getProfileAndReinitialize(item);
});
```